Formula/p/postgis-pg17.rb

```diff
@@ -0,0 +1,150 @@
+class PostgisPg17 < Formula
```

Comment:
Wasn't sure on name if we go this route, e.g.
* `postgis-postgresql16` may be like some other formulae (e.g. `qt5`), but a bit lengthy.
* `postgis-pg16` is an abbreviated variant, which is reverse of MacPorts (i.e. `pg16-postgis`)
* `postgis16` could be an option like GTK dependents (where we often add `3` or `4`), but then it looks more like part of actual name (e.g. `libxml2`) which is confusing.

---

NOTE: Probably will end up dropping most of these commits when stable PG 17 is released. Changes are mostly to test that they work for most of our dependents